contracts/base/KeeperCompatibleHarvester.sol

```diff
@@ -0,0 +1,545 @@
+    uint256 public _performUpkeepGasLimitBuffer;
+    uint256 public _vaultHarvestFunctionGasOverhead; // Estimated average gas cost of calling harvest()
+    uint256 public _keeperRegistryGasOverhead; // Gas cost of upstream contract that calls performUpkeep(). This is a private variable on KeeperRegistry.
+    uint256 public _chainlinkUpkeepTxPremiumFactor; // Tx premium factor/multiplier scaled by 1 gwei (10**9).
+    address public _callFeeRecipient;
+
+    // State variables that will change across upkeeps.
+    uint256 public _startIndex;
+
+    constructor(
+        address keeperRegistry_,
+        uint256 performUpkeepGasLimit_,
+        uint256 performUpkeepGasLimitBuffer_,
+        uint256 vaultHarvestFunctionGasOverhead_,
+        uint256 keeperRegistryGasOverhead_
+    ) {
+        // Set contract references.
+        _keeperRegistry = IKeeperRegistry(keeperRegistry_);
+
+        // Initialize state variables from initialize() arguments.
+        _performUpkeepGasLimit = performUpkeepGasLimit_;
+        _performUpkeepGasLimitBuffer = performUpkeepGasLimitBuffer_;
+        _vaultHarvestFunctionGasOverhead = vaultHarvestFunctionGasOverhead_;
+        _keeperRegistryGasOverhead = keeperRegistryGasOverhead_;
+
+        // Initialize state variables derived from initialize() arguments.
+        (uint32 paymentPremiumPPB, , , , , , ) = _keeperRegistry.getConfig();
+        _chainlinkUpkeepTxPremiumFactor = uint256(paymentPremiumPPB);
+    }
+
+    /*             */
+    /* checkUpkeep */
+    /*             */
+
+    function checkUpkeep(
+        bytes calldata checkData_ // unused
+    )
+        external
+        view
+        override
+        returns (
+            bool upkeepNeeded_,
+            bytes memory performData_ // array of vaults +
+        )
+    {
+        checkData_; // dummy reference to get rid of unused parameter warning
+
+        // get vaults to iterate over
+        address[] memory vaults = _getVaultAddresses();
+
+        // count vaults to harvest that will fit within gas limit
+        (
+            HarvestInfo[] memory harvestInfo,
+            uint256 numberOfVaultsToHarvest,
+            uint256 newStartIndex
+        ) = _countVaultsToHarvest(vaults);
+        if (numberOfVaultsToHarvest == 0)
+            return (false, bytes("BeefyAutoHarvester: No vaults to harvest"));
+
+        (
+            address[] memory vaultsToHarvest,
+            uint256 heuristicEstimatedTxCost,
+            uint256 callRewards
+        ) = _buildVaultsToHarvest(vaults, harvestInfo, numberOfVaultsToHarvest);
+
+        uint256 nonHeuristicEstimatedTxCost = _calculateExpectedTotalUpkeepTxCost(
+                numberOfVaultsToHarvest
+            );
+
+        performData_ = abi.encode(
+            vaultsToHarvest,
+            newStartIndex,
+            heuristicEstimatedTxCost,
+            nonHeuristicEstimatedTxCost,
+            callRewards
+        );
+
+        return (true, performData_);
```

Comment:
```suggestion
        if (vaultsToHarvest.length > 0)
            return (true, performData_);
        else
            return (false, performData_);
```
Maybe will be better if we check the data here instead in performUpkeep function which send and reports and that will be expensive?

Reply:
It's already checked: https://github.com/hackbg/chainlink-keeper-templates/blob/feature/abstract-harvester/contracts/base/KeeperCompatibleHarvester.sol#L75